ktor-http/js/src/io/ktor/http/URLBuilderJs.kt

```diff
@@ -10,17 +10,23 @@ import io.ktor.util.*
  * Hostname of current origin.
  *
  * It uses "localhost" for all platforms except js.
+ *
+ * Note that not all platforms have location set. React Native platofrms expose window without a location.
  */
 public actual val URLBuilder.Companion.origin: String
     get() = when (PlatformUtils.platform) {
         Platform.Browser -> {
             js(
                 """
-                var origin = ""
+                var tmpLocation = null
                 if (typeof window !== 'undefined') {
-                  origin = window.location.origin 
+                  tmpLocation = window.location
                 } else {
-                  origin = self.location.origin 
+                  tmpLocation = self.location
+                }
+                var origin = ""
+                if (tmpLocation) {
+                  origin = location.origin
```

Comment:
Should it be `tmpLocation` instead of `location` here?  If yes, why call it `tmpLocation`?

Reply:
Do you also mind checking that `self` is defined above? This will fix https://youtrack.jetbrains.com/issue/KTOR-5978

Reply:
> Should it be tmpLocation instead of location here?

Yes, indeed it should. `location` worked in my testing because `self` and `window` both represent global scope, so you can reference their members directly.

> If yes, why call it `tmpLocation`?

This tripped me up! At first I tried with code like:

```
                if (typeof window !== 'undefined') {
                  location = window.location
                } else {
                  location = self.location
                }
```

But in the browser, when you set `location = window.location` it is cute little shortcut to reload the page!!!

> Do you also mind checking that self is defined above? 

Done!